### DISCUSSION TITLE
Routing directly to detail page of new created plant

### DISCUSSION THREAD

**Discussion by SabinePfanner** [neuefische/questions#844] [Web] [answered]
We've created a form which enables the user to create a new plant, using the `POST` method. The form is rendered on a specific page `/create` and the `POST` method works correctly. 

But after the plant has been successfully created, the user should be redirected to the details page of the NEW created prop. Currently the user is routed back to the homepage.

Any idea, how we can redirect the user to the details page of the new created crop (`/plants/$[id]`) instead of back to the homepage (`/`)?

``` 
 async function addPlant(plant) {
    const response = await fetch(`/api/plants/`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(plant),
    });
    if (!response.ok) {
      console.error(response.status);
      return;
    }
    if (response.ok) {
      console.log(response.status);
      mutate();
      router.push(`/`);
    }
  }
```

Current branch: [feature-create-plants](https://github.com/SabinePfanner/plants-app/tree/feature-create-plants)

**Comment by doemser** [comment, marked as answer]
This should probably solve it:

`/api/plants/index.js`

```javascript
const newPlant = await Plant.create(plantData);
response.status(201).json({ message: "New crop successfully created", id: newPlant.id });
```

> Not sure if newPlant.id is correct, you need to recheck that

Then in your POST fetch go:

```javascript
router.push(`/plants/${response.id}`);
```

**Reply by SabinePfanner** [reply]
Thanks, now it's working!
